### DISCUSSION TITLE
How to get original request body from response?

### DISCUSSION THREAD

**Discussion by spasquali-ww** [sindresorhus/got#1537] [Q&A] [answered]
I'd like to get back the original request `body` on a POST `response`. Is that possible?

Note that I'm setting `throwHttpErrors` to `false`, so parsing through this mixed `response`.

I'm able to find the Symbol for (body) in `response.value.request` and derive what I need from there, but am hoping there is a shortcut (don't want to iterate through `response` Symbols on every request)

Thanks!

- [x ] I have read the documentation.

**Comment by szmarczak** [comment, marked as answer]
`response.request.options.body` but note that it won't work for streams.

**Reply by rightaway** [reply]
On `HTTPError` I'm getting undefined for `e.response.request.options.body` even though `Symbol(body)` is under `e.response.request` with the request content. Is `options.body` not there in all cases? It's not a stream I'm using just a regular http request.

**Reply by szmarczak** [reply]
@rightaway We have `options.json` and `options.form` as well.

**Comment by spasquali-ww** [comment]
Just what I needed (`response.value.request.options.body` in my case). Thank you so much!
